library/azure_rm_trafficmanager.py

```diff
@@ -0,0 +1,588 @@
+    target_resource_id=dict(type='str'),
+    target=dict(type='str'),
+    endpoint_status=dict(type='str'),
+    weight=dict(type='int'),
+    priority=dict(type='int'),
+    endpoint_location=dict(type='str'),
+    endpoint_monitor_status=dict(type='str'),
+    min_child_endpoints=dict(type='int'),
+    geo_mapping=dict(type='list', elements='str')
+)
+
+
+class AzureRMTrafficManager(AzureRMModuleBase):
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str'
+            ),
+            profile_status=dict(
+                type='str',
+                default='Enabled',
+                choices=['Enabled', 'Disabled']
+            ),
+            traffic_routing_method=dict(
+                type='str',
+                default='Performance',
+                choices=['Performance', 'Priority', 'Weighted', 'Geographic']
+            ),
+            dns_config=dict(
+                type='dict',
+                options=dns_config_spec
+            ),
+            monitor_config=dict(
+                type='dict',
+                default=dict(
+                    protocol='http',
+                    port=80,
+                    path='/'
+                ),
+                options=monitor_config_spec
+            ),
+            endpoints=dict(
+                type='list',
+                elements='dict',
+                options=endpoint_spec,
+                default=[]
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.tags = None
+        self.location = None
+        self.profile_status = None
+        self.traffic_routing_method = None
+        self.dns_config = None
+        self.monitor_config = None
+        self.endpoints = None
+
+        required_if = [
+            ('state', 'present', [])
+        ]
+
+        self.results = dict(
+            changed=False,
+            state=dict()
+        )
+
+        super(AzureRMTrafficManager, self).__init__(derived_arg_spec=self.module_arg_spec, supports_check_mode=True, required_if=required_if)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        to_be_updated = False
+
+        if not self.dns_config:
+            self.dns_config = dict(
+                relative_name=self.name,
+                ttl=60
+            )
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
```

Comment:
if self.location doesn't inherit from resource_group, remove the redundant `get_resource_group` call